blog/posts/swiip/the-weight-of-js.md

```diff
@@ -0,0 +1,191 @@
+  - treeshaking
+  - lazyloading
+---
+
+There are two principal measures for JS performance. The one which is focusing all the attention is DOM updating which is almost maxed out by recent frameworks. It is also really dependant of the application code and often not a real need of your application (who cares about milliseconds of rendering performances when your backend spent 2s for gathering your data?).
+
+The other one, frameworks often don't even communicate on real figures, hide themselves behind features supposed to resolve the question like SSR, Tree Shaking or Lazy Loading. Yet, in many cases, it's a way more important performance issue than any other question of your JS app.
+
+> It is the weight of your JavaScript code.
+
+Never lose the thought that even if we see today websites as fully featured apps. It's still a platform where the whole app is downloaded, compiled and booted when you open your tab. Imagine that for Outlook, Word or Photoshop!
+
+A webapp loading starts with the network bandwidth to download the file then the CPU to uncompress and compile it. The more code you got the most your app will need bandwith and CPU. Even then, more code will eventually mean a slower app. Look at this tweet from Nintendo.
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">Switch OS UI Session: in cutting player&#39;s wait time as much as possible, Switch&#39;s home menu design resources have less than 200KB. amazing. <a href="https://t.co/gnINFXnFPp">https://t.co/gnINFXnFPp</a></p>&mdash; Takashi Mochizuki (@mochi_wsj) <a href="https://twitter.com/mochi_wsj/status/1032276091279302656?ref_src=twsrc%5Etfw">August 22, 2018</a></blockquote>
+
+They have almost no network consideration to deal with (as far as I know) yet they care deeply at their app size for performances and fluidity.
+
+Lastly: "less code is less bug". I always loved this quotation. Shipping less code (both yours and third parties) will always be a way to a better stability.
+
+## Tools don't solve everything
+
+A good part of the so rich Web tooling we use nowadays is about limiting the size of your JS. All of these techniques are optimizations which is great and I recommend to use almost all of them. Yet, optimizations are not solutions and the real amount of JS is still a question.
+
+### Bundling
+
+The first tool we use about JS performance is bundling. We don't serve the browser as many files as we develop knowing that every request to the server has its cost. Most of the time, we compile a bundle of one, two to five bundled JS file. Yet, you still has as much code as before.
+
+### Minifying
+
+Also for a long time, we use JS minification. It means to modify the code by removing most of the spaces, line breaks, rename variables to shorter names to have the same code with less characters and then, a smaller weight. It's great, it has almost no counterpart unless readability and then debuggability but it can be handled by the use of source maps.
```

Comment:
```suggestion
For a long time, we've been using JS minification, which means to modify the code by removing most of the spaces, line breaks, renaming variables to shorter names to have the same code with less characters and then, a smaller weight. It's great, it has almost no downside except readability and then debuggability but it can be handled by the use of source maps.
```